include/circt/Dialect/HW/HWAttributes.td

```diff
@@ -313,4 +313,47 @@ def EnumFieldAttr : AttrDef<HWDialect, "EnumField"> {
   }];
 }
 
+let cppNamespace = "circt::hw" in {
+  // Define enum for all the different waivers.
+  def NoAssignXWaiver : I32EnumAttrCase<"NoAssignX", 0>;
+  def ExcludeCoverageWaiver : I32EnumAttrCase<"ExcludeCoverage", 1>;
+  // Kind of Waiver
+  def WaiverKindAttr : I32EnumAttr<"WaiverKind", "waiver kind",
+      [NoAssignXWaiver, ExcludeCoverageWaiver]> ;
+}
+
+// An attribute to indicate the lint waivers that need to be emitted for the
+// module.
+def WaiverAttr : AttrDef<HWDialect, "Waiver"> {
```

Comment:
If this attribute is specific to module, can we change the name to `WaiverAttr`?